lib/core/src/sdk.rs

```diff
@@ -1187,9 +1299,45 @@ impl LiquidSdk {
     ) -> Result<ReceivePaymentResponse, PaymentError> {
         self.ensure_is_started().await?;
 
-        let payer_amount_sat = req.prepare_res.payer_amount_sat;
-        let fees_sat = req.prepare_res.fees_sat;
+        let PrepareReceivePaymentResponse {
+            payer_amount_sat,
+            use_lightning,
+            fees_sat,
+        } = req.prepare_response;
+
+        match (use_lightning, payer_amount_sat) {
+            (true, None) => {
+                return Err(PaymentError::Generic {
+                    err: "Payer amount was not set.".to_string(),
+                });
+            }
+            (true, Some(payer_amount_sat)) => {
+                self.create_receive_swap(payer_amount_sat, fees_sat, req.description.clone())
+                    .await
+            }
+            (false, None) => {
+                let address = self.onchain_wallet.next_unused_address().await?.to_string();
+                Ok(ReceivePaymentResponse {
+                    receive_destination: ReceiveDestination::Liquid { address },
+                })
+            }
+            (false, Some(payer_amount_sat)) => {
+                let address = self.onchain_wallet.next_unused_address().await?.to_string();
+                let uri =
+                    utils::new_liquid_bip21(&address, self.config.network, Some(payer_amount_sat));
```

Comment:
Should we add the description as "message" or "label" in the resulted bip21 ?

Reply:
I believe it should be the message field, the label is meant for small strings/identifiers as far as I understand